ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -0,0 +1,147 @@
+        )
+
+    def _set_base_span_tags(
+        self,
+        span: Span,
+        model: Optional[str] = None,
+        api_key: Optional[str] = None,
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """Set base level tags that should be present on all Anthropic spans (if they are not None)."""
+        if model is not None:
+            span.set_tag_str(MODEL, model)
+        if api_key is not None:
+            if len(api_key) >= 4:
+                span.set_tag_str(API_KEY, f"...{str(api_key[-4:])}")
+            else:
+                span.set_tag_str(API_KEY, api_key)
+
+    def _extract_input_message(self, messages):
+        """Extract input messages from the stored prompt.
+        Anthropic allows for messages and multiple texts in a message, which requires some special casing.
+        """
+        if not isinstance(messages, Iterable):
+            log.warning("Anthropic input must be a list of messages.")
+
+        input_messages = []
+        for message in messages:
+            if not isinstance(message, dict):
+                log.warning("Anthropic message input must be a list of message param dicts.")
+                continue
+
+            content = message.get("content", None)
+            role = message.get("role", None)
+
+            if role is None or content is None:
+                log.warning("Anthropic input message must have content and role.")
+
+            if isinstance(content, str):
+                input_messages.append({"content": self.trunc(content), "role": role})
+
+            elif isinstance(content, list):
+                for entry in content:
+                    if entry.get("type") == "text":
+                        input_messages.append({"content": self.trunc(entry.get("text", "")), "role": role})
+                    elif entry.get("type") == "image":
+                        # Store a placeholder for potentially enormous binary image data.
+                        input_messages.append({"content": "([IMAGE DETECTED])", "role": role})
+                    else:
+                        input_messages.append({"content": str(entry), "role": role})
+
+        return input_messages
+
+    def _extract_output_message(self, response):
+        """Extract output messages from the stored response."""
+        output_messages = []
+        content = _get_attr(response, "content", None)
+        role = _get_attr(response, "role", "")
+
+        if isinstance(content, str):
+            return [{"content": self.trunc(content), "role": role}]
+
+        elif isinstance(content, list):
+            for completion in content:
+                text = _get_attr(completion, "text", None)
+                if isinstance(text, str):
+                    output_messages.append({"content": self.trunc(text), "role": role})
+        return output_messages
+
+    def record_usage(self, span: Span, usage: Dict[str, Any]) -> None:
+        if not usage or not self.metrics_enabled:
```

Comment:
```suggestion
        if not usage:
```
We don't need to worry about dogstatsd metrics for anthropic